README.md

```diff
@@ -1,2 +1,104 @@
-# styleguide
-contains runnable coding guidelines to maintain consistency across all our repos  
+# Runnable Code and Test Patterns Guide
+
+Runnable's approach to writing code and tests.
+
+## Table of Contents
+
+1. [Code Formatting](#code-formatting)
+1. [Testing](#testing)
+    1. [Types of Tests](#types-of-tests)
+    1. [Stubbing](#stubbing)
+    1. [Asserting](#asserting)
+    1. [Additional Testing Reading](#additional-testing-reading)
+
+## Code Formatting
+
+[![js-standard-style](https://cdn.rawgit.com/feross/standard/master/badge.svg)](https://github.com/feross/standard)
+
+We use [`standard`](http://standardjs.com/) in most projects. If it passes `standard`, anything else is up to the developer. No questions asked.
+
+**[back to the top](#table-of-contents)**
+
+## Testing
+
+Let's discuss various tests, what they should do and cover, and how best to organize them.
+
+### Types of Tests
+
+There are three types of tests we should be concerned about.
+
+- Unit
+- Integration
+- Functional
+
+#### Definitions
+
+**Unit** tests are pretty obvious (or should be). They test exactly one unit of functionality. Most likely this should follow the boundaries of functions or at largest, modules. If we're finding we're writing a lot unit tests for a specific function, it may need to be broken up.
+
+An important note: unit tests should _never_ touch an external service (e.g. database).
```

Comment:
For the most part, @thejsj, you don't ever have to test that `mongoose` is doing the right thing (though I question it daily). Nothing there really needs to be tested. For the various other models where we are making requests through the database, we're usually manipulating data somehow and then sending that data into `mongoose` for it to do the work on the database.

In a **unit** test, mocking out the layer right at `mongoose` and simply returning sample objects is sufficient. If there's more paculiar logic that you may not be sure of or is more complex (I'm thinking like doing a map-reduce in mongo or making sure `populate` is doing the correct thing), you would move to an **integration** test and write it there.

The goal with **unit** tests is to be completely offline and not require _any_ external dependency. **Integration** tests start to bring in external dependencies and **functional** tests will start the entire server/project.